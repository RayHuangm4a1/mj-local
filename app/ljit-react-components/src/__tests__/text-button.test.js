import React from 'react';
import { shallow, mount, } from 'enzyme';
import TextButton from '../components/text-button';

describe('TextButton', () => {
	it('handle default props', () => {
		const {
			color,
			disabled,
			onClick,
			fontSize,
			fontWeight,
		} = TextButton.defaultProps;

		expect(color).toEqual('default');
		expect(disabled).toEqual(false);
		expect(fontSize).toEqual('medium');
		expect(fontWeight).toEqual('normal');
		expect(onClick).toBeDefined();
		expect(onClick).toBeInstanceOf(Function);
	});

	it('should renders correctly', () => {
		const disabled = false;
		const className = 'mock-button';
		const text = 'mock-text';
		const fontSize = 'medium';
		const fontWeight = 'normal';
		const wrapper = shallow(
			<TextButton
				disabled={disabled}
				className={className}
				text={text}
				fontSize={fontSize}
				fontWeight={fontWeight}
			/>
		);

		expect(wrapper).toMatchSnapshot();
	});

	it('should be selectable by class ljit-text-button', () => {
		const wrapper = shallow(<TextButton/>);

		expect(wrapper.hasClass('ljit-text-button')).toEqual(true);
	});

	it('should be selectable by custom class', () => {
		const className = 'mock-class';
		const text ='mock-text';
		const wrapper = shallow(
			<TextButton
				className={className}
				text={text}
			/>
		);

		expect(wrapper.hasClass(className)).toEqual(true);
	});

	it('should be selectable by ljit-text-button--default if props.color equal default ', () => {
		const wrapper = shallow(
			<TextButton color={'default'}/>);

		expect(wrapper.hasClass('ljit-text-button--default')).toEqual(true);
	});

	it('should be selectable by ljit-text-button--danger if props.color equal default ', () => {
		const wrapper = shallow(
			<TextButton color={'danger'}/>);

		expect(wrapper.hasClass('ljit-text-button--danger')).toEqual(true);
	});
	it('should be selectable by ljit-text-button--text-small if props.fontSize equal small ', () => {
		const wrapper = shallow(
			<TextButton fontSize={'small'}/>);

		expect(wrapper.hasClass('ljit-text-button--text-small')).toEqual(true);
	});
	it('should be selectable by ljit-text-button--text-medium if props.fontSize equal medium ', () => {
		const wrapper = shallow(
			<TextButton fontSize={'medium'}/>);

		expect(wrapper.hasClass('ljit-text-button--text-medium')).toEqual(true);
	});
	it('should be selectable by ljit-text-button--weight-normal if props.fontWeight equal normal ', () => {
		const wrapper = shallow(
			<TextButton fontWeight={'normal'}/>);

		expect(wrapper.hasClass('ljit-text-button--weight-normal')).toEqual(true);
	});
	it('should be selectable by ljit-text-button--weight-heavy if props.fontWeight equal heavy ', () => {
		const wrapper = shallow(
			<TextButton fontWeight={'heavy'}/>);

		expect(wrapper.hasClass('ljit-text-button--weight-heavy')).toEqual(true);
	});

	it('should mount in a full DOM', () => {
		const disabled = false;
		const className = 'mock-class';
		const text ='mock-text';
		const color = 'default';
		const onClick = () => {};

		const wrapper = mount(
			<TextButton
				disabled={disabled}
				color={color}
				className={className}
				onClick={onClick}
				text={text}
			/>
		);

		expect(wrapper.props().disabled).toEqual(disabled);
		expect(wrapper.props().color).toBe(color);
		expect(wrapper.props().className).toBe(className);
		expect(wrapper.props().onClick).toEqual(onClick);
		expect(wrapper.props().text).toBe(text);

	});

	it('should handle onclick event', () => {
		const onClick = jest.fn();
		const wrapper = shallow(<TextButton onClick={onClick} />);

		wrapper.simulate('click');
		expect(onClick).toHaveBeenCalledTimes(1);
		expect(onClick).toHaveBeenCalled();
	});
});

describe('TextButton SizeEnums ', () => {
	it('should contains SMALL property', () => {
		const typeName = 'small';
		const formatType = 'SMALL';

		expect(TextButton.SizeEnums[formatType]).toEqual(typeName);
	});

	it('should contains MEDIUM property', () => {
		const typeName = 'medium';
		const formatType = 'MEDIUM';

		expect(TextButton.SizeEnums[formatType]).toEqual(typeName);
	});
});
describe('TextButton WeightEnums ', () => {
	it('should contains NORMAL property', () => {
		const typeName = 'normal';
		const formatType = 'NORMAL';

		expect(TextButton.WeightEnums[formatType]).toEqual(typeName);
	});

	it('should contains HEAVY property', () => {
		const typeName = 'heavy';
		const formatType = 'HEAVY';

		expect(TextButton.WeightEnums[formatType]).toEqual(typeName);
	});
});
