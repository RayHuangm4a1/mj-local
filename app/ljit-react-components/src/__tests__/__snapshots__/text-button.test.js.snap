// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextButton should renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextButton
    className="mock-button"
    color="default"
    disabled={false}
    fontSize="medium"
    fontWeight="normal"
    onClick={[Function]}
    text="mock-text"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "mock-text",
      "className": "ljit-text-button ljit-text-button--default ljit-text-button--text-medium ljit-text-button--weight-normal mock-button",
      "disabled": false,
      "onClick": [Function],
      "type": "button",
    },
    "ref": null,
    "rendered": "mock-text",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "mock-text",
        "className": "ljit-text-button ljit-text-button--default ljit-text-button--text-medium ljit-text-button--weight-normal mock-button",
        "disabled": false,
        "onClick": [Function],
        "type": "button",
      },
      "ref": null,
      "rendered": "mock-text",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
